capz/run-capz-e2e.sh

```diff
@@ -181,6 +182,254 @@ run_e2e_test() {
+        log "VFP tag count: $count"
+        if [[ $count != $ipsetCount ]]; then
+            log "WARNING: VFP tag count $count does not match HNS IPSET count $ipsetCount"
+            hadFailure=true
+        fi
+    done
+    if [[ $hadEndpoints == false ]]; then
+        log "WARNING: VFP tags not validated for NPM since no endpoints found on node $npmNode"
+    fi
+    if [[ $hadFailure == true ]]; then
+        log "ERROR: VFP tags are inconsistent with HNS SetPolicies"
+        capture_npm_hns_state
+        return 1
+    fi
+}
+
+# results in a file called npm-hns-state.zip
+capture_npm_hns_state () {
+    log "capturing NPM HNS state..."
+    kubectl get pod -owide -A
+    test -d npm-hns-state/ && rm -rf npm-hns-state/ || true
+    mkdir npm-hns-state
+    cd npm-hns-state
+    curl -LO https://raw.githubusercontent.com/Azure/azure-container-networking/master/debug/windows/npm/win-debug.sh
+    chmod u+x ./win-debug.sh
+    curl -LO https://raw.githubusercontent.com/Azure/azure-container-networking/master/debug/windows/npm/pod_exec.ps1
+    ./win-debug.sh
+    cd ..
+    zip -9qr npm-hns-state.zip npm-hns-state
+    # to unzip:
+    # unzip npm-hns-state.zip -d npm-hns-state
+}
+
+# currently takes ~3 hours to run
+# e.g. 19:37:05 to 22:32:44 and 19:16:18 to 22:29:13
+run_npm_conformance () {
+    ## install NPM e2e binary
+    log "ensuring NPM e2e binary is installed"
+    rc=0; test -f npm-e2e.test || rc=$?
+    if [[ $rc == 0 ]]; then
+        log "NPM e2e binary found, skipping install"
+    else
+        log "NPM e2e binary not found, installing..."
+        test -d npm-kubernetes/ && rm -rf npm-kubernetes/ || true
+        mkdir npm-kubernetes
+        cd npm-kubernetes
+        # NOTE: if this is not downloaded every run, then probably need to sleep before the VFP tag verification
+        git clone https://github.com/huntergregory/kubernetes.git --depth=1 --branch=quit-on-failure
+        cd kubernetes
+        make WHAT=test/e2e/e2e.test
+        cd ../..
+        mv npm-kubernetes/kubernetes/_output/local/bin/linux/amd64/e2e.test ./npm-e2e.test
+        rm -rf npm-kubernetes/
+    fi
+
+    log "beginning npm conformance test..."
+
+    toRun="NetworkPolicy"
+
+    nomatch1="should enforce policy based on PodSelector or NamespaceSelector"
+    nomatch2="should enforce policy based on NamespaceSelector with MatchExpressions using default ns label"
+    nomatch3="should enforce policy based on PodSelector and NamespaceSelector"
+    nomatch4="should enforce policy based on Multiple PodSelectors and NamespaceSelectors"
+    cidrExcept1="should ensure an IP overlapping both IPBlock.CIDR and IPBlock.Except is allowed"
+    cidrExcept2="should enforce except clause while egress access to server in CIDR block"
+    namedPorts="named port"
+    wrongK8sVersion="Netpol API"
+    toSkip="\[LinuxOnly\]|$nomatch1|$nomatch2|$nomatch3|$nomatch4|$cidrExcept1|$cidrExcept2|$namedPorts|$wrongK8sVersion|SCTP"
```

Comment:
I think we'll want these to be local to NPM, because this code block will just validate NetworkPolicy tests for NPM on the forked e2e binary.

Perhaps we could have a `NPM_GINKGO_SKIP` if we want to configure?